src/Client.ts

```diff
@@ -201,11 +216,13 @@ export default class Client {
   }
 
   // publish the key bundle into the contact topic
-  private async publishUserContact(): Promise<void> {
-    const pub = this.keys.getPublicKeyBundle()
+  // WARNING: temporarily public to allow testing negotiated topics
+  // TODO: make private again asap
+  async publishUserContact(legacy = true): Promise<void> {
+    const keys = legacy ? this.legacyKeys : this.keys
     await this.publishEnvelope({
       contentTopic: buildUserContactTopic(this.address),
-      message: pub.toBytes(),
+      message: encodeContactBundle(keys.getPublicKeyBundle()),
```

Comment:
Apparently we've still been publishing bare key bundle instead of contact bundle.